main/src/ca/uwaterloo/flix/tools/pkg/Trust.scala

```diff
@@ -16,7 +16,72 @@
  */
 package ca.uwaterloo.flix.tools.pkg
 
-sealed trait Trust
+sealed trait Trust {
+
+  /**
+    * Combines `this` and `other` according to the following lattice:
+    * {{{
+    *     Unrestricted
+    *          |
+    *   TrustJavaClass
+    *          |
+    *        Plain
+    * }}}
+    */
+  def glb(other: Trust): Trust = (this, other) match {
+    // TODO: Add tests
+    case (Trust.Plain, Trust.Plain) => Trust.Plain
+    case (Trust.Plain, Trust.TrustJavaClass) => Trust.Plain
+    case (Trust.Plain, Trust.Unrestricted) => Trust.Plain
+    case (Trust.TrustJavaClass, Trust.Plain) => Trust.Plain
+    case (Trust.TrustJavaClass, Trust.TrustJavaClass) => Trust.TrustJavaClass
+    case (Trust.TrustJavaClass, Trust.Unrestricted) => Trust.TrustJavaClass
+    case (Trust.Unrestricted, Trust.Plain) => Trust.Plain
+    case (Trust.Unrestricted, Trust.TrustJavaClass) => Trust.TrustJavaClass
+    case (Trust.Unrestricted, Trust.Unrestricted) => Trust.Unrestricted
+  }
+
+  def lub(other: Trust): Trust = {
+    if (this.glb(other) != this) {
+      this
+    } else {
+      other
+    }
+  }
+
+  /**
+    * Returns `true` if `this` is less than or equal to `other` in the lattice
+    * {{{
+    *     Unrestricted
+    *          |
+    *   TrustJavaClass
+    *          |
+    *        Plain
+    * }}}
+    */
+  def lessThanEq(other: Trust): Boolean = (this, other) match {
+    // TODO: Add tests
+    case (Trust.Plain, Trust.Plain) => true
+    case (Trust.Plain, Trust.TrustJavaClass) => true
+    case (Trust.Plain, Trust.Unrestricted) => true
+    case (Trust.TrustJavaClass, Trust.Plain) => false
+    case (Trust.TrustJavaClass, Trust.TrustJavaClass) => true
+    case (Trust.TrustJavaClass, Trust.Unrestricted) => true
+    case (Trust.Unrestricted, Trust.Plain) => false
+    case (Trust.Unrestricted, Trust.TrustJavaClass) => false
+    case (Trust.Unrestricted, Trust.Unrestricted) => true
+  }
+
+  /**
+    * Returns `true` iff `!this.lessThanOrEq(other)` holds.
+    *
+    * @see [[lessThanEq]]
+    */
+  def greaterThan(other: Trust): Boolean = {
+    // TODO: Add tests
+    !this.lessThanEq(other)
+  }
+}
```

Comment:
I don't think you need all lattice operations.

Note that if you define lub, you can also defined <= in terms of it.

Allows a very short and obviously correct implementation.